hydrolib/core/io/fnm/models.py

```diff
@@ -0,0 +1,231 @@
+    boundaries: Path = Path("pluvius.dwa")
+    pluvius: Path = Path("pluvius.3b")
+    pluvius_general: Path = Path("pluvius.alg")
+    kasklasse: Path = Path("kasklass")
+    bui_file: Path = Path("default.bui")
+    verdampings_file: Path = Path("default.evp")
+    unpaved_area_general: Path = Path("unpaved.3b")
+    unpaved_area_storage: Path = Path("unpaved.sto")
+    green_house_area_initialisation: Path = Path("kasinit")
+    green_house_area_usage_data: Path = Path("kasgebr")
+    crop_factors: Path = Path("cropfact")
+    table_bergingscoef: Path = Path("bergcoef")
+    unpaved_alpha_factor_definitions: Path = Path("unpaved.alf")
+    run_log_file: Path = Path("sobek_3b.log")
+    schema_overview: Path = Path("3b_gener.out")
+    output_results_paved_area: Path = Path("paved.out")
+    output_results_unpaved_area: Path = Path("unpaved.out")
+    output_results_green_houses: Path = Path("grnhous.out")
+    output_results_open_water: Path = Path("openwate.out")
+    output_results_structures: Path = Path("struct3b.out")
+    output_results_boundaries: Path = Path("bound3b.out")
+    output_results_pluvius: Path = Path("pluvius.out")
+    infiltration_definitions_unpaved: Path = Path("unpaved.inf")
+    run_debug_file: Path = Path("sobek_3b.dbg")
+    unpaved_seepage: Path = Path("unpaved.sep")
+    unpaved_initial_values_tables: Path = Path("unpaved.tbl")
+    green_house_general: Path = Path("greenhse.3b")
+    green_house_roof_storage: Path = Path("greenhse.rf")
+    pluvius_sewage_entry: Path = Path("runoff.out")
+    input_variable_gauges_on_edge_nodes: Path = Path("sbk_rtc.his")
+    input_salt_data: Path = Path("salt.3b")
+    input_crop_factors_open_water: Path = Path("crop_ow.prn")
+    restart_file_input: Path = Path("RSRR_IN")
+    restart_file_output: Path = Path("RSRR_OUT")
+    binary_file_input: Path = Path("3b_input.bin")
+    sacramento_input: Path = Path("sacrmnto.3b")
+    output_flow_rates_edge_nodes: Path = Path("aanvoer.abr")
+    output_salt_concentration_edge: Path = Path("saltbnd.out")
+    output_salt_exportation: Path = Path("salt.out")
+    greenhouse_silo_definitions: Path = Path("greenhse.sil")
+    open_water_general: Path = Path("openwate.3b")
+    open_water_seepage_definitions: Path = Path("openwate.sep")
+    open_water_tables_target_levels: Path = Path("openwate.tbl")
+    structure_general: Path = Path("struct3b.dat")
+    structure_definitions: Path = Path("struct3b.def")
+    controller_definitions: Path = Path("contr3b.def")
+    structure_tables: Path = Path("struct3b.tbl")
+    boundary_data: Path = Path("bound3b.3b")
+    boundary_tables: Path = Path("bound3b.tbl")
+    sobek_location_rtc: Path = Path("sbk_loc.rtc")  # TODO 58.
+    wwtp_data: Path = Path("wwtp.3b")
+    wwtp_tables: Path = Path("wwtp.tbl")
+    industry_general: Path = Path("industry.3b")
+
+    mappix_paved_area_sewage_storage_name: Literal["pvstordt.his"] = "pvstordt.his"
```

Comment:
With some more reading, what about `Literal[Path("pvstordt.his")]`? 🤯

Reply:
I have added the additional comment  to explain the reasoning for using Literals.

Reply:
@BeardedPlatypus 

Because *all* these properties are Paths (right?), this can work.

```python
from pydantic import BaseModel
from pydantic import validator
from typing import Literal
from pathlib import Path



class A(BaseModel):
    a: Literal[Path("test.tif")]

    @validator("*", pre=True)
    def make_path(cls, v):
        if isinstance(v, str):
            return Path(v)
        return v


A(a=Path("test.tif"))  # works
A(a="test.tif")  # works
A(a="test.tif2")  # fails
```